### PR TITLE
disable_autorebuild_by_default

### PR DESCRIPTION
**Pull request by rcerven**
raise error if autorebuild aren't explicitly set

### CODE REVIEW

File: atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py [hunk 1]
```diff
@@ -52,13 +52,16 @@ def _is_rebuild_enabled(self):
...
+
             except ValueError as e:
-                self.log.error('can\'t parse [autorebuild].enabled as bool in "%s", assuming'
-                               'autorebuild is enabled (error: "%s")', self.config_file, e)
+                self.log.error('can\'t parse [autorebuild].enabled as bool in "%s", failing'
+                               'plugin (error: "%s")', self.config_file, e)
+                raise AutoRebuildCanceledException(self.key, 'can\'t parse "%s"' % config_file)
         else:
-            self.log.info('no "%s", assuming autorebuild is enabled', self.config_file)
+            self.log.info('no "%s", failing plugin', self.config_file)
```
Comment (vrutkovs) [inline atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py]: Nitpick: this probably should be a `warn` instead

File: atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py [hunk 2]
```diff
@@ -52,13 +52,16 @@ def _is_rebuild_enabled(self):
                               'enabled' if result else 'disabled',
                               self.config_file)
             except configparser.Error:
-                self.log.error('can\'t parse "%s", assuming autorebuild is enabled',
-                               self.config_file)
+                self.log.error('can\'t parse "%s", failing plugin', self.config_file)
+                raise AutoRebuildCanceledException(self.key, 'can\'t parse "%s"' % config_file)
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py]: This method shouldn't be raising exceptions. It should return a bool to indicate whether or not rebuild is enabled.

File: atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py [hunk 3]
```diff
@@ -52,13 +52,16 @@ def _is_rebuild_enabled(self):
                               'enabled' if result else 'disabled',
                               self.config_file)
             except configparser.Error:
-                self.log.error('can\'t parse "%s", assuming autorebuild is enabled',
-                               self.config_file)
+                self.log.error('can\'t parse "%s", failing plugin', self.config_file)
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py]: These should be: ... "assuming autorebuild is disabled."

File: atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py [hunk 4]
```diff
@@ -52,13 +52,16 @@ def _is_rebuild_enabled(self):
                               'enabled' if result else 'disabled',
                               self.config_file)
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py]: If you initialize `result` to `False`, you only need to adjust the log messages and set it to `True` here.

File: tests/plugins/test_stop_autorebuild_if_disabled.py [hunk 5]
```diff
@@ -118,17 +118,18 @@ def test_malformed_config(self, caplog):
         # flexmock(configparser.SafeConfigParser).should_receive('getboolean').\
         #     with_args('autorebuild', 'enabled').and_raise(configparser.Error)
         # assert this doesn't raise
```
Comment (twaugh) [inline tests/plugins/test_stop_autorebuild_if_disabled.py]: This comment is no longer correct. While here you could remove the commented-out code if that's no longer relevant.

File: tests/plugins/test_stop_autorebuild_if_disabled.py [hunk 6]
```diff
@@ -118,17 +118,18 @@ def test_malformed_config(self, caplog):
...
         self.assert_message_logged(
-            'can\'t parse ".osbs-repo-config", assuming autorebuild is enabled',
+            'can\'t parse ".osbs-repo-config", assuming autorebuild is disabled',
             caplog)
 
     def test_no_config(self, caplog):
         if MOCK:
             mock_docker()
 
         # assert this doesn't raise
```
Comment (twaugh) [inline tests/plugins/test_stop_autorebuild_if_disabled.py]: Comment needs updating.

File: atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py (hunk 4 above)
Comment (twaugh) [inline atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py]: Where is result set to `True`? More importantly, if it is never set to `True`, how come the unit testing doesn't catch that?

**Comment by rcerven** [comment]
line 50:  result = cfp.getboolean('autorebuild', 'enabled')

File: atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py (hunk 4 above)
Comment (twaugh) [inline atomic_reactor/plugins/pre_stop_autorebuild_if_disabled.py]: Ah, line 50, thanks.

**Comment by lcarva** [comment]
@rcerven please squash and fix commit message

**Comment by rcerven** [comment]
squashed and changed commit message

**Pull request merged**